docs/zh/week06/06-3.md

```diff
@@ -0,0 +1,436 @@
+
+与堆叠形式的输入同理。
+
+<!--$y[t]$ is calculated at the final rotation and then we can use the chain rule to backpropagate the error to the previous time step.-->
+
+$y[t]$由最后一步的旋转计算得出，之后我们可以用链式法则来反向传递前一步的残差。
+
+
+### 语言建模中的分批处理（Batch-Ification in Language Modeling ）
+
+<!--When dealing with a sequence of symbols, we can batchify the text into different sizes. For example, when dealing with sequences shown in the following figure, batch-ification can be applied first, where the time domain is preserved vertically. In this case, the batch size is set to 4.-->
+
+在处理符号序列时，我们可以将文字分批成不同的大小。比如，在处理图18中的序列时，先将它分批处理（Batch-Ification），它的时间定义域是垂直向。这个例子中的Batch的大小为4。
+
+<center>
+<img src="{{site.baseurl}}/images/week06/06-3/batchify_1.png" style="zoom: 40%; background-color:#DCDCDC;"/><br>
+<b>图18:</b> Batch-Ification
+</center>
+
+<!--If BPTT period $T$ is set to 3, the first input $x[1:T]$ and output $y[1:T]$ for RNN is determined as-->
+
+如果将BPTT的周期$T$设为3， 那么第一组RNN的输入$x[1:T]$和输出$y[1:T]$为：
+$$
+\begin{aligned}
+x[1:T] &= \begin{bmatrix}
+a & g & m & s \\
+b & h & n & t \\
+c & i & o & u \\
+\end{bmatrix} \\
+y[1:T] &= \begin{bmatrix}
+b & h & n & t \\
+c & i & o & u \\
+d & j & p & v
+\end{bmatrix}
+\end{aligned}
+$$
+
+<!--When performing RNN on the first batch, firstly, we feed $x[1] = [a\ g\ m\ s]$ into RNN and force the output to be $y[1] = [b\ h\ n\ t]$. The hidden representation $h[1]$ will be sent forward into next time step to help the RNN predict $y[2]$ from $x[2]$. After sending $h[T-1]$ to the final set of $x[T]$ and $y[T]$, we cut gradient propagation process for both $h[T]$ and $h[0]$ so that gradients will not propagate infinitely(.detach() in Pytorch). The whole process is shown in figure below.-->
+
+在运行RNN的第一组样本时，我们首先引入 $x[1] = [a\ g\ m\ s]$并强制输出为$y[1] = [b\ h\ n\ t]$。隐藏层表达式$h[1]$向前传递到下一步，帮助RNN从$x[2]$预测$y[2]$。当传递$h[T-1]$至最后一组$x[T]$和$y[T]$后，我们停止$h[T]$和$h[0]$的梯度传递步骤，使得梯度不会无限地传递（.detach() in Pytorch）。整个过程如图19所示。
+
+<center>
+<img src="{{site.baseurl}}/images/week06/06-3/batchify_2.png" style="zoom: 40%; background-color:#DCDCDC;"/><br>
+<b>图19:</b> Batch-Ification
+</center>
+
+
+
+## 梯度消失与梯度爆炸（Vanishing and Exploding Gradient）
+
+### <!--Problem--> 
+
+### 问题
+
+<center>
+<img src="{{site.baseurl}}/images/week06/06-3/rnn_3.png" style="zoom: 40%; background-color:#DCDCDC;"/><br>
+<b>图20:</b> 梯度消失问题
+</center>
+
+<!--The figure above is a typical RNN architecture. In order to perform rotation over previous steps in RNN, we use matrices, which can be regarded as horizontal arrows in the model above. Since the matrices can change the size of outputs, if the determinant we select is larger than 1, the gradient will inflate over time and cause gradient explosion. Relatively speaking, if the eigenvalue we select is small across 0, the propagation process will shrink gradients and leads to the gradient vanishing.-->
+
+图20展示了一个经典RNN架构。我们用矩阵对RNN里的前一步进行旋转操作，在这个模型中用水平箭头表示。假设我们选择的行列式大于1，因为矩阵可以改变输出的大小，梯度会随时间增大并造成梯度爆炸。相对而言，如果我们选择的特征值小于0，则传播过程会收缩梯度并导致梯度消失。
```

Comment:
Just need to confirm: "the eigenvalue is small across 0" == "the eigenvalue is smaller than 0"?

Reply:
Yeah, it should be "smaller than zero".